FiligranHQ/xtm-hub
# [backend/frontend] fix(organization): check if an organization already exists with same name (case insensitive)

# Context: 
The purpose of this PR is to fix an issue where an organization with the same name (case insensitive) could be created despite a database constraint. PostgreSQL is case sensitive, so the constraint in the database wasn't sufficient to prevent such duplicates. To address this, I added a case-insensitive check in the code to ensure no duplicates are allowed.

Additionally, I moved the error message handling to the backend, so that translations can be properly managed on the frontend.

# How to test:  

1. Pull the branch locally.
2. Attempt to create an organization with a name that already exists in the database but with a different case (e.g., `MyOrganization` and `myorganization`).
3. Verify that the system prevents the creation of the duplicate organization with a proper error message.
4. Ensure that organizations with unique names can still be created successfully.
5. Verify that the error message is displayed correctly on the frontend with the appropriate translation.


# What tests has been made: 

# Additional information: 
N/A

close #135
